File: abc/322/C++/b.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n, m;
    string s, t;
    cin >> n >> m >> s >> t;
    int ans = 0;
    if (!t.starts_with(s)) ans += 2;
    if (!t.ends_with(s)) ans++;
    cout << ans << endl;
    return 0;
}

File: abc/322/Python/b.py
def main():
    n, m = map(int, input().split())
    s = input()
    t = input()

    if t.startswith(s) and t.endswith(s):
        return 0
    elif t.startswith(s) and not t.endswith(s):
        return 1
    elif t.endswith(s) and not t.startswith(s):
        return 2
    return 3

if __name__ == '__main__':
    print(main())